src/transformers/agents/python_interpreter.py

```diff
@@ -440,73 +483,71 @@ def evaluate_name(name, state, tools):
     raise InterpreterError(f"The variable `{name.id}` is not defined.")
 
 
-def evaluate_condition(condition, state, tools):
-    left = evaluate_ast(condition.left, state, tools)
-    comparators = [evaluate_ast(c, state, tools) for c in condition.comparators]
+def evaluate_condition(condition, state, static_tools, custom_tools):
+    left = evaluate_ast(condition.left, state, static_tools, custom_tools)
+    comparators = [evaluate_ast(c, state, static_tools, custom_tools) for c in condition.comparators]
     ops = [type(op) for op in condition.ops]
 
     result = True
     current_left = left
 
     for op, comparator in zip(ops, comparators):
         if op == ast.Eq:
-            result = result and (current_left == comparator)
```

Comment:
This change allows to support pandas "==" statement between a Series and an int for instance